dashboard/app/models/user.rb

```diff
@@ -804,12 +802,20 @@ def self.new_with_session(params, session)
     end
   end
 
+  def managing_own_account?
+    provider.blank? || (provider == User::PROVIDER_MANUAL)
+  end
+
   def password_required?
-    # password is required if:
-    (!persisted? || # you are a new user
-     !password.nil? || !password_confirmation.nil?) && # or changing your password
-      (provider.blank? || (User::PROVIDER_MANUAL == provider)) # and you are a person creating your own account
-    # (as opposed to a person who had their account created for them or are logging in with oauth)
+    # Password is not required if the user is not managing their own account
+    # (i.e., someone is creating their account for them or the user is using OAuth).
+    return false unless managing_own_account?
```

Comment:
I'm a little worried this name is long-term ambiguous - it'd be easy to mix up "managing own account" with concepts of "personal account" or "account ownership" which is not what we're talking about here.  Maybe `managing_own_credentials` is better?